### PR TITLE
Move Engine State from EngineQueue to EngineController

### PR DESCRIPTION
**Pull request by trianglesphere**
**Description**

This moves all the control of the execution engine from the EngineQueue struct to the EngineController struct. The Engine Controller remains in the derive package for now to minimize the amount of changes in this PR. The EngineQueue continues to implement the EngineControl interface & forwards all methods to the EngineController which contains the actual implementation.

**Tests**

No tests as this should be pure tidying.

### CODE REVIEW

**Comment by axelKingsley** [comment]
Per PR Description

> No tests as this should be pure tidying.

Shouldn't the unit tests *also* move to the new location (Engine Controller)? If unit tests live on the old structure, but that structure now proxies its implementation down to the EC, the unit tests do still cover the implementation, but only indirectly. Moving them would help us more clearly check the behavior of the underlying EC, especially if we some day wanted to implement more than one type of EC.

(I am looking around now and thinking maybe there just aren't tests to move at this time)

**Comment by trianglesphere** [comment]
@axelKingsley I looked at the coverage. For the most part we only testing the happy path. I'd like to split it out / update the interface a bit more before doing EngineController testing.

**Comment by axelKingsley** [comment]
That's totally fine regarding when to prioritize testing refactors. Thanks for looking it over! 

(I wish GitHub allowed direct response to top level comments...)

File: op-node/rollup/derive/engine_controller.go [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+	if err != nil {
+		return errTyp, err
+	}
+
+	e.buildingID = id
+	e.buildingSafe = updateSafe
+	e.buildingOnto = parent
+	if updateSafe {
+		e.safeAttrs = attrs
+	}
```
Comment (ajsutton) [inline op-node/rollup/derive/engine_controller.go:139]: This wasn't in the old `StartPayload` from engine queue. It looks to me like the logic for when to update this is correct but it's the one noticeable logic change in this that's not just moving code around.

Given we only just started building the new payload should this variable be called `buildingSafeAttrs` or something to indicate they aren't (yet) the safe attrs? Admittedly it was `safeAttributes` in the original so probably not a big deal.

File: op-node/rollup/derive/engine_queue.go [hunk 2]
```diff
@@ -683,94 +680,30 @@ func (eq *EngineQueue) forceNextSafeAttributes(ctx context.Context) error {
...
-
-	if eq.buildingSafe {
-		eq.pendingSafeHead = ref
-		if updateSafe {
-			eq.safeHead = ref
-			eq.postProcessSafeL2()
-			eq.metrics.RecordL2Ref("l2_safe", ref)
-		}
+	postSafe := eq.ec.SafeL2Head()
+	if buildingSafe && (preSafe != postSafe) {
```
Comment (ajsutton) [inline op-node/rollup/derive/engine_queue.go:695]: Would it be better to having `ConfirmPayload` return a boolean to indicate if it changed the safe head? (I could probably go either way on this but figured its worth asking).

**Review by ajsutton — APPROVED** [review]
LGTM.

File: op-node/rollup/derive/engine_controller.go (hunk 1 above)
Comment (trianglesphere) [reply op-node/rollup/derive/engine_controller.go:139]: Good callout that was an addition to match the behavior of the engine queue (we need to track if we are building a span batch or not). Ideally in a follow up we'd change the API to be like `InsertDerivedBlocked(attributes eth.(..), isSpanBatch, isLastBatchInSpan bool)`

File: op-node/rollup/derive/engine_queue.go (hunk 2 above)
Comment (trianglesphere) [reply op-node/rollup/derive/engine_queue.go:695]: It'd be nice, but I wanted to limit the amount of changes I was making at once. The first attempt of changing the API at the same time as moving everything around was a lot more painful.

Comment (ajsutton) [reply op-node/rollup/derive/engine_queue.go:695]: Yep, makes sense - I'm ok with improving things step by step.

File: op-node/rollup/derive/engine_controller.go (hunk 1 above)
Comment (ajsutton) [reply op-node/rollup/derive/engine_controller.go:139]: Ok, it's not bad as-is so happy to leave it and then hopefully the whole API will improve over time.

File: op-node/rollup/derive/engine_queue.go [hunk 3]
```diff
@@ -100,21 +127,7 @@ type EngineQueue struct {
 	log log.Logger
 	cfg *rollup.Config
 
-	finalized  eth.L2BlockRef
-	safeHead   eth.L2BlockRef
-	unsafeHead eth.L2BlockRef
-
-	// L2 block processed from the batch, but not consolidated to the safe block yet.
```
Comment (pcw109550) [inline op-node/rollup/derive/engine_queue.go:107]: We have lost this comment while refactoring.

**Pull request merged**